ibis/pyspark/compiler.py

```diff
@@ -154,8 +155,13 @@ def compile_cast(t, expr, scope, **kwargs):
                 'in the PySpark backend. {} not allowed.'.format(type(op.arg))
             )
 
+    if isinstance(op.to, dtypes.Array):
+        cast_type = ibis_array_dtype_to_spark_dtype(op.to)
+    else:
+        cast_type = op.to.name
```

Comment:
Can we use some ibis_dtypr->spark_dtype function? I don't think using `to.name` is very safe - it happens that the string representation of ibis and spark are the same, but this feels a bit fragile